Litmusbook for elasticsearch-statefulset deployment.

vibhor995 opened the pull request:
Signed-off-by: vibhor995 <vibhor.kumar@mayadata.io>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:

1) This PR consist litmusbook  deployment of Elasticsearch statefulset

**Special notes for your reviewer**:
@dargasudarshan @gprasath @nsathyaseelan 


```No config file found; using defaults
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
 [WARNING]: Found variable using reserved name: action

PLAY [localhost] ***************************************************************
2019-01-08T07:49:04.635312 (delta: 0.042758)         elapsed: 0.042758 ******** 
=============================================================================== 

TASK [Gathering Facts] *********************************************************
2019-01-08T07:49:04.645602 (delta: 0.010255)         elapsed: 0.053048 ******** 
ok: [127.0.0.1]

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:08.155555 (delta: 3.509877)         elapsed: 3.563001 ******** 
included: /common/utils/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
2019-01-08T07:49:08.204516 (delta: 0.048924)         elapsed: 3.611962 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
2019-01-08T07:49:08.245445 (delta: 0.04089)         elapsed: 3.652891 ********* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:08.285384 (delta: 0.039866)         elapsed: 3.69283 ********* 
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
2019-01-08T07:49:08.348569 (delta: 0.063143)         elapsed: 3.756015 ******** 
changed: [127.0.0.1] => {"changed": true, "checksum": "6527a253a359ed86afebe9a5acb7cef1cdd5ceca", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "64d3f3b6a9cafa1258bf4b48430a64d0", "mode": "0644", "owner": "root", "size": 425, "src": "/root/.ansible/tmp/ansible-tmp-1546933748.39-170762281062485/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T07:49:08.818968 (delta: 0.470354)         elapsed: 4.226414 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.363653", "end": "2019-01-08 07:49:09.407855", "rc": 0, "start": "2019-01-08 07:49:09.044202", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: elasticsearch-deployment \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: elasticsearch-deployment ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T07:49:09.444530 (delta: 0.62552)         elapsed: 4.851976 ********* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.820105", "end": "2019-01-08 07:49:10.389396", "failed_when_result": false, "rc": 0, "start": "2019-01-08 07:49:09.569291", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/elasticsearch-deployment configured", "stdout_lines": ["litmusresult.litmus.io/elasticsearch-deployment configured"]}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
2019-01-08T07:49:10.430926 (delta: 0.986357)         elapsed: 5.838372 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T07:49:10.466193 (delta: 0.035227)         elapsed: 5.873639 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T07:49:10.503548 (delta: 0.037252)         elapsed: 5.910994 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:10.542877 (delta: 0.039264)         elapsed: 5.950323 ******** 
included: /common/utils/pre_create_app_deploy.yml for 127.0.0.1

TASK [Check whether the provider storageclass is applied] **********************
2019-01-08T07:49:10.615659 (delta: 0.072739)         elapsed: 6.023105 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get sc openebs-cstor-sparse", "delta": "0:00:00.463744", "end": "2019-01-08 07:49:11.222301", "rc": 0, "start": "2019-01-08 07:49:10.758557", "stderr": "", "stderr_lines": [], "stdout": "NAME                   PROVISIONER                    AGE\nopenebs-cstor-sparse   openebs.io/provisioner-iscsi   47d", "stdout_lines": ["NAME                   PROVISIONER                    AGE", "openebs-cstor-sparse   openebs.io/provisioner-iscsi   47d"]}

TASK [Replace the pvc placeholder with provider] *******************************
2019-01-08T07:49:11.258507 (delta: 0.642806)         elapsed: 6.665953 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
2019-01-08T07:49:11.524783 (delta: 0.266233)         elapsed: 6.932229 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:11.694337 (delta: 0.169512)         elapsed: 7.101783 ******** 
included: /common/utils/fetch_replica_values.yml for 127.0.0.1

TASK [Get the application replica values from env] *****************************
2019-01-08T07:49:11.760000 (delta: 0.065631)         elapsed: 7.167446 ******** 
ok: [127.0.0.1] => {"ansible_facts": {"app_rkey": "replicas", "app_rvalue": "1"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
2019-01-08T07:49:11.825420 (delta: 0.065378)         elapsed: 7.232866 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application label values from env] *******************************
2019-01-08T07:49:11.989992 (delta: 0.164463)         elapsed: 7.397438 ******** 
ok: [127.0.0.1] => {"ansible_facts": {"app_lkey": "app", "app_lvalue": "elasticsearch"}, "changed": false}

TASK [Replace the application label placeholder in deployment spec] ************
2019-01-08T07:49:12.054048 (delta: 0.063987)         elapsed: 7.461494 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "3 replacements made"}

TASK [Enable/Disable I/O based liveness probe] *********************************
2019-01-08T07:49:12.220433 (delta: 0.166344)         elapsed: 7.627879 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:12.266025 (delta: 0.04555)         elapsed: 7.673471 ********* 
included: /common/utils/create_ns.yml for 127.0.0.1

TASK [Obtain list of existing namespaces] **************************************
2019-01-08T07:49:12.338515 (delta: 0.072376)         elapsed: 7.745961 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get ns --no-headers -o custom-columns=:metadata.name", "delta": "0:00:00.458998", "end": "2019-01-08 07:49:12.925394", "rc": 0, "start": "2019-01-08 07:49:12.466396", "stderr": "", "stderr_lines": [], "stdout": "app-esearch-ns\napp-jenkins-ns\ndefault\nkube-public\nkube-system\nlitmus\nopenebs\ntest", "stdout_lines": ["app-esearch-ns", "app-jenkins-ns", "default", "kube-public", "kube-system", "litmus", "openebs", "test"]}

TASK [Create test specific namespace.] *****************************************
2019-01-08T07:49:12.964027 (delta: 0.625409)         elapsed: 8.371473 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:13.010499 (delta: 0.04643)         elapsed: 8.417945 ********* 
included: /common/utils/status_testns.yml for 127.0.0.1

TASK [Checking the status  of test specific namespace.] ************************
2019-01-08T07:49:13.075171 (delta: 0.06463)         elapsed: 8.482617 ********* 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get ns app-esearch-ns -o custom-columns=:status.phase --no-headers", "delta": "0:00:00.451461", "end": "2019-01-08 07:49:13.672314", "rc": 0, "start": "2019-01-08 07:49:13.220853", "stderr": "", "stderr_lines": [], "stdout": "Active", "stdout_lines": ["Active"]}

TASK [Replace the application label placeholder in rbac spec] ******************
2019-01-08T07:49:13.712515 (delta: 0.637289)         elapsed: 9.119961 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "3 replacements made"}

TASK [Replace the namespace placeholder in rbac spec.] *************************
2019-01-08T07:49:13.880568 (delta: 0.16794)         elapsed: 9.288014 ********* 
changed: [127.0.0.1] => {"changed": true, "msg": "3 replacements made"}

TASK [Deploying Elasticsearch rbac.] *******************************************
2019-01-08T07:49:14.048805 (delta: 0.168196)         elapsed: 9.456251 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f esearch_rbac.yml", "delta": "0:00:00.710276", "end": "2019-01-08 07:49:14.887191", "rc": 0, "start": "2019-01-08 07:49:14.176915", "stderr": "", "stderr_lines": [], "stdout": "serviceaccount/elasticsearch-logging created\nclusterrole.rbac.authorization.k8s.io/elasticsearch-logging configured\nclusterrolebinding.rbac.authorization.k8s.io/elasticsearch-logging configured", "stdout_lines": ["serviceaccount/elasticsearch-logging created", "clusterrole.rbac.authorization.k8s.io/elasticsearch-logging configured", "clusterrolebinding.rbac.authorization.k8s.io/elasticsearch-logging configured"]}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:14.922481 (delta: 0.873635)         elapsed: 10.329927 ******* 
included: /common/utils/deploy_single_app.yml for 127.0.0.1

TASK [Deploying Elasticsearch] *************************************************
2019-01-08T07:49:14.992625 (delta: 0.070104)         elapsed: 10.400071 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f es-statefulset.yaml -n app-esearch-ns", "delta": "0:00:00.548722", "end": "2019-01-08 07:49:15.704095", "rc": 0, "start": "2019-01-08 07:49:15.155373", "stderr": "", "stderr_lines": [], "stdout": "statefulset.apps/elasticsearch-logging created", "stdout_lines": ["statefulset.apps/elasticsearch-logging created"]}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:15.742037 (delta: 0.749373)         elapsed: 11.149483 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T07:49:15.793620 (delta: 0.051539)         elapsed: 11.201066 ******* 
included: /common/utils/check_replica_count.yml for 127.0.0.1

TASK [Obtain the number of replicas.] ******************************************
2019-01-08T07:49:15.877823 (delta: 0.084156)         elapsed: 11.285269 ******* 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get statefulset -n app-esearch-ns -l \"app=elasticsearch\" -o custom-columns=:spec.replicas", "delta": "0:00:00.535150", "end": "2019-01-08 07:49:16.629048", "rc": 0, "start": "2019-01-08 07:49:16.093898", "stderr": "", "stderr_lines": [], "stdout": "\n1", "stdout_lines": ["", "1"]}

TASK [Obtain the ready replica count and compare with the replica count.] ******
2019-01-08T07:49:16.669956 (delta: 0.792086)         elapsed: 12.077402 ******* 
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (30 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (29 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (28 retries left).
changed: [127.0.0.1] => {"attempts": 4, "changed": true, "cmd": "kubectl get statefulset -n app-esearch-ns -l \"app=elasticsearch\" -o custom-columns=:..readyReplicas", "delta": "0:00:00.456977", "end": "2019-01-08 07:52:19.535053", "rc": 0, "start": "2019-01-08 07:52:19.078076", "stderr": "", "stderr_lines": [], "stdout": "\n1", "stdout_lines": ["", "1"]}

TASK [Deprovisioning the Application] ******************************************
2019-01-08T07:52:19.577774 (delta: 182.907734)         elapsed: 194.98522 ***** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [set_fact] ****************************************************************
2019-01-08T07:52:19.622244 (delta: 0.044303)         elapsed: 195.02969 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [include_tasks] ***********************************************************
2019-01-08T07:52:19.672839 (delta: 0.050553)         elapsed: 195.080285 ****** 
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
2019-01-08T07:52:19.727070 (delta: 0.054192)         elapsed: 195.134516 ****** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T07:52:19.763107 (delta: 0.036)         elapsed: 195.170553 ********* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T07:52:19.798483 (delta: 0.035338)         elapsed: 195.205929 ****** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
2019-01-08T07:52:19.833649 (delta: 0.035122)         elapsed: 195.241095 ****** 
changed: [127.0.0.1] => {"changed": true, "checksum": "4da6298723b4228f5cad23ca2155b6dd5f813feb", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "5822a4984246284d298ea2bfb437d54f", "mode": "0644", "owner": "root", "size": 423, "src": "/root/.ansible/tmp/ansible-tmp-1546933939.88-63053314367412/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T07:52:20.841346 (delta: 1.00766)         elapsed: 196.248792 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.369406", "end": "2019-01-08 07:52:21.343034", "rc": 0, "start": "2019-01-08 07:52:20.973628", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: elasticsearch-deployment \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Pass ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: elasticsearch-deployment ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Pass "]}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T07:52:21.393111 (delta: 0.551726)         elapsed: 196.800557 ****** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.556782", "end": "2019-01-08 07:52:22.080324", "failed_when_result": false, "rc": 0, "start": "2019-01-08 07:52:21.523542", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/elasticsearch-deployment configured", "stdout_lines": ["litmusresult.litmus.io/elasticsearch-deployment configured"]}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=32   changed=19   unreachable=0    failed=0   

2019-01-08T07:52:22.107318 (delta: 0.714083)         elapsed: 197.514764 ****** 
=============================================================================== ```

dargasudarshan commented:
@vibhor995 Can you paste the ansible test logs?

gprasath commented on apps/elasticsearch/deployers/es-statefulset.yaml:
```diff
@@ -0,0 +1,117 @@
+subjects:
+- kind: ServiceAccount
+  name: elasticsearch-logging
+  namespace: logging
```
Why different namespace is being used here?

vibhor995 replied:
@gprasath i will remove this namespaces becouse we are already deploying it on certain namespaces.....looks like different services are also having seperate namespaces..

gprasath commented on apps/elasticsearch/deployers/run_litmus_test.yml:
```diff
@@ -0,0 +1,54 @@
+            
+          # Use 'deprovision' for app-clean up
+          - name: ACTION
+            value: provision
```
Can you test the deletion also and paste the output.

vibhor995 replied:
@gprasath yes i have tested this for deprovisioning also in both  jiva and cstor data plains.i will paste the output.

vibhor995 commented:
@gprasath **deprovision output**
```No config file found; using defaults
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
 [WARNING]: Found variable using reserved name: action

PLAY [localhost] ***************************************************************
2019-01-08T08:41:08.556261 (delta: 0.038803)         elapsed: 0.038803 ******** 
=============================================================================== 

TASK [Gathering Facts] *********************************************************
2019-01-08T08:41:08.566647 (delta: 0.010308)         elapsed: 0.049189 ******** 
ok: [127.0.0.1]

TASK [include_tasks] ***********************************************************
2019-01-08T08:41:12.066552 (delta: 3.499794)         elapsed: 3.549094 ******** 
included: /common/utils/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
2019-01-08T08:41:12.137120 (delta: 0.070476)         elapsed: 3.619662 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
2019-01-08T08:41:12.204803 (delta: 0.067637)         elapsed: 3.687345 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T08:41:12.247159 (delta: 0.042231)         elapsed: 3.729701 ******** 
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
2019-01-08T08:41:12.307772 (delta: 0.060561)         elapsed: 3.790314 ******** 
changed: [127.0.0.1] => {"changed": true, "checksum": "6527a253a359ed86afebe9a5acb7cef1cdd5ceca", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "64d3f3b6a9cafa1258bf4b48430a64d0", "mode": "0644", "owner": "root", "size": 425, "src": "/root/.ansible/tmp/ansible-tmp-1546936872.35-53947141721062/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T08:41:12.801831 (delta: 0.494013)         elapsed: 4.284373 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.368164", "end": "2019-01-08 08:41:13.411277", "rc": 0, "start": "2019-01-08 08:41:13.043113", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: elasticsearch-deployment \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: elasticsearch-deployment ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T08:41:13.447209 (delta: 0.645269)         elapsed: 4.929751 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.939388", "end": "2019-01-08 08:41:14.533151", "failed_when_result": false, "rc": 0, "start": "2019-01-08 08:41:13.593763", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/elasticsearch-deployment configured", "stdout_lines": ["litmusresult.litmus.io/elasticsearch-deployment configured"]}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
2019-01-08T08:41:14.576186 (delta: 1.128935)         elapsed: 6.058728 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T08:41:14.612603 (delta: 0.036377)         elapsed: 6.095145 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T08:41:14.647718 (delta: 0.035047)         elapsed: 6.13026 ********* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T08:41:14.684123 (delta: 0.036338)         elapsed: 6.166665 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Replace the application label placeholder in rbac spec] ******************
2019-01-08T08:41:14.729785 (delta: 0.045621)         elapsed: 6.212327 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Replace the namespace placeholder in rbac spec.] *************************
2019-01-08T08:41:14.773649 (delta: 0.043573)         elapsed: 6.256191 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Deploying Elasticsearch rbac.] *******************************************
2019-01-08T08:41:14.820112 (delta: 0.046348)         elapsed: 6.302654 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-01-08T08:41:14.865202 (delta: 0.045045)         elapsed: 6.347744 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Deprovisioning the Application] ******************************************
2019-01-08T08:41:14.909855 (delta: 0.044605)         elapsed: 6.392397 ******** 
included: /common/utils/deprovision_statefulset.yml for 127.0.0.1

TASK [Check if the statefulset application exists.] ****************************
2019-01-08T08:41:15.020718 (delta: 0.110746)         elapsed: 6.50326 ********* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n app-esearch-ns -l app=elasticsearch", "delta": "0:00:00.473535", "end": "2019-01-08 08:41:15.643285", "failed_when_result": false, "rc": 0, "start": "2019-01-08 08:41:15.169750", "stderr": "", "stderr_lines": [], "stdout": "NAME                      READY   STATUS    RESTARTS   AGE\nelasticsearch-logging-0   1/1     Running   0          52m", "stdout_lines": ["NAME                      READY   STATUS    RESTARTS   AGE", "elasticsearch-logging-0   1/1     Running   0          52m"]}

TASK [Obtaining PVCs related to the application.] ******************************
2019-01-08T08:41:15.684038 (delta: 0.663279)         elapsed: 7.16658 ********* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc -n app-esearch-ns -l app=elasticsearch --no-headers -o custom-columns=:.metadata.name", "delta": "0:00:00.464305", "end": "2019-01-08 08:41:16.293861", "rc": 0, "start": "2019-01-08 08:41:15.829556", "stderr": "", "stderr_lines": [], "stdout": "openebs-esearch-elasticsearch-logging-0", "stdout_lines": ["openebs-esearch-elasticsearch-logging-0"]}

TASK [Obtaining the PV names.] *************************************************
2019-01-08T08:41:16.332274 (delta: 0.648189)         elapsed: 7.814816 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc -l app=elasticsearch -n app-esearch-ns --no-headers -o custom-columns=:.spec.volumeName", "delta": "0:00:00.463593", "end": "2019-01-08 08:41:16.941453", "rc": 0, "start": "2019-01-08 08:41:16.477860", "stderr": "", "stderr_lines": [], "stdout": "pvc-e581a3ae-1319-11e9-b4e3-42010a800142", "stdout_lines": ["pvc-e581a3ae-1319-11e9-b4e3-42010a800142"]}

TASK [Check for presence & value of cas type annotation] ***********************
2019-01-08T08:41:16.978537 (delta: 0.646214)         elapsed: 8.461079 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pv pvc-e581a3ae-1319-11e9-b4e3-42010a800142 --no-headers -o jsonpath=\"{.metadata.annotations.openebs\\\\.io/cas-type}\"", "delta": "0:00:00.465717", "end": "2019-01-08 08:41:17.584067", "rc": 0, "start": "2019-01-08 08:41:17.118350", "stderr": "", "stderr_lines": [], "stdout": "cstor", "stdout_lines": ["cstor"]}

TASK [Record the storage engine name] ******************************************
2019-01-08T08:41:17.621412 (delta: 0.642834)         elapsed: 9.103954 ******** 
ok: [127.0.0.1] => {"ansible_facts": {"stg_engine": "cstor"}, "changed": false}

TASK [Replace the PVC name in application deployer spec.] **********************
2019-01-08T08:41:17.685566 (delta: 0.064076)         elapsed: 9.168108 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
2019-01-08T08:41:17.976713 (delta: 0.291063)         elapsed: 9.459255 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application replica values from env] *****************************
2019-01-08T08:41:18.157763 (delta: 0.181006)         elapsed: 9.640305 ******** 
ok: [127.0.0.1] => {"ansible_facts": {"app_rkey": "replicas", "app_rvalue": "1"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
2019-01-08T08:41:18.233424 (delta: 0.075617)         elapsed: 9.715966 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application label values from env] *******************************
2019-01-08T08:41:18.422966 (delta: 0.189497)         elapsed: 9.905508 ******** 
ok: [127.0.0.1] => {"ansible_facts": {"app_lkey": "app", "app_lvalue": "elasticsearch"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
2019-01-08T08:41:18.492617 (delta: 0.069604)         elapsed: 9.975159 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "3 replacements made"}

TASK [Delete the application and its related service.] *************************
2019-01-08T08:41:18.679557 (delta: 0.186895)         elapsed: 10.162099 ******* 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl delete -f es-statefulset.yaml -n app-esearch-ns", "delta": "0:00:00.469320", "end": "2019-01-08 08:41:19.292261", "rc": 0, "start": "2019-01-08 08:41:18.822941", "stderr": "", "stderr_lines": [], "stdout": "statefulset.apps \"elasticsearch-logging\" deleted", "stdout_lines": ["statefulset.apps \"elasticsearch-logging\" deleted"]}

TASK [Deleting the PVC] ********************************************************
2019-01-08T08:41:19.336437 (delta: 0.656831)         elapsed: 10.818979 ******* 
changed: [127.0.0.1] => (item=openebs-esearch-elasticsearch-logging-0) => {"changed": true, "cmd": "kubectl delete pvc openebs-esearch-elasticsearch-logging-0 -n app-esearch-ns", "delta": "0:00:00.487489", "end": "2019-01-08 08:41:19.975228", "item": "openebs-esearch-elasticsearch-logging-0", "rc": 0, "start": "2019-01-08 08:41:19.487739", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"openebs-esearch-elasticsearch-logging-0\" deleted", "stdout_lines": ["persistentvolumeclaim \"openebs-esearch-elasticsearch-logging-0\" deleted"]}

TASK [Check if the PVCs are deleted] *******************************************
2019-01-08T08:41:20.017120 (delta: 0.680637)         elapsed: 11.499662 ******* 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pvc -n app-esearch-ns", "delta": "0:00:00.461123", "end": "2019-01-08 08:41:20.653158", "rc": 0, "start": "2019-01-08 08:41:20.192035", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [Check if the ctrl-pod is deleted.] ***************************************
2019-01-08T08:41:20.693445 (delta: 0.676275)         elapsed: 12.175987 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Check if the  replica-pod is deleted.] ***********************************
2019-01-08T08:41:20.742223 (delta: 0.048704)         elapsed: 12.224765 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Check if the target pod is deleted.] *************************************
2019-01-08T08:41:20.790146 (delta: 0.047805)         elapsed: 12.272688 ******* 
FAILED - RETRYING: Check if the target pod is deleted. (20 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (19 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (18 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (17 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (16 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (15 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (14 retries left).
FAILED - RETRYING: Check if the target pod is deleted. (13 retries left).
changed: [127.0.0.1] => (item=pvc-e581a3ae-1319-11e9-b4e3-42010a800142) => {"attempts": 9, "changed": true, "cmd": "kubectl get pods -l openebs.io/persistent-volume=pvc-e581a3ae-1319-11e9-b4e3-42010a800142 -n openebs --no-headers", "delta": "0:00:00.444473", "end": "2019-01-08 08:42:06.209756", "item": "pvc-e581a3ae-1319-11e9-b4e3-42010a800142", "rc": 0, "start": "2019-01-08 08:42:05.765283", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [debug] *******************************************************************
2019-01-08T08:42:06.255000 (delta: 45.464808)         elapsed: 57.737542 ****** 
ok: [127.0.0.1] => {
    "msg": {
        "changed": true, 
        "msg": "All items completed", 
        "results": [
            {
                "_ansible_ignore_errors": null, 
                "_ansible_item_label": "pvc-e581a3ae-1319-11e9-b4e3-42010a800142", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "attempts": 9, 
                "changed": true, 
                "cmd": "kubectl get pods -l openebs.io/persistent-volume=pvc-e581a3ae-1319-11e9-b4e3-42010a800142 -n openebs --no-headers", 
                "delta": "0:00:00.444473", 
                "end": "2019-01-08 08:42:06.209756", 
                "failed": false, 
                "invocation": {
                    "module_args": {
                        "_raw_params": "kubectl get pods -l openebs.io/persistent-volume=pvc-e581a3ae-1319-11e9-b4e3-42010a800142 -n openebs --no-headers", 
                        "_uses_shell": true, 
                        "argv": null, 
                        "chdir": null, 
                        "creates": null, 
                        "executable": "/bin/bash", 
                        "removes": null, 
                        "stdin": null, 
                        "warn": true
                    }
                }, 
                "item": "pvc-e581a3ae-1319-11e9-b4e3-42010a800142", 
                "rc": 0, 
                "start": "2019-01-08 08:42:05.765283", 
                "stderr": "No resources found.", 
                "stderr_lines": [
                    "No resources found."
                ], 
                "stdout": "", 
                "stdout_lines": []
            }
        ]
    }
}

TASK [Check status of the pool deployments from cvr] ***************************
2019-01-08T08:42:06.323792 (delta: 0.068755)         elapsed: 57.806334 ******* 
changed: [127.0.0.1] => (item=pvc-e581a3ae-1319-11e9-b4e3-42010a800142) => {"attempts": 1, "changed": true, "cmd": "kubectl get cvr -n openebs -l openebs.io/persistent-volume=pvc-e581a3ae-1319-11e9-b4e3-42010a800142 --no-headers", "delta": "0:00:00.454554", "end": "2019-01-08 08:42:06.933617", "item": "pvc-e581a3ae-1319-11e9-b4e3-42010a800142", "rc": 0, "start": "2019-01-08 08:42:06.479063", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [Delete the namespace.] ***************************************************
2019-01-08T08:42:06.979967 (delta: 0.656136)         elapsed: 58.462509 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl delete ns app-esearch-ns", "delta": "0:00:05.965875", "end": "2019-01-08 08:42:13.085103", "rc": 0, "start": "2019-01-08 08:42:07.119228", "stderr": "", "stderr_lines": [], "stdout": "namespace \"app-esearch-ns\" deleted", "stdout_lines": ["namespace \"app-esearch-ns\" deleted"]}

TASK [set_fact] ****************************************************************
2019-01-08T08:42:13.122484 (delta: 6.142479)         elapsed: 64.605026 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [include_tasks] ***********************************************************
2019-01-08T08:42:13.175554 (delta: 0.053024)         elapsed: 64.658096 ******* 
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
2019-01-08T08:42:13.234618 (delta: 0.059013)         elapsed: 64.71716 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T08:42:13.271788 (delta: 0.037091)         elapsed: 64.75433 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T08:42:13.307508 (delta: 0.035674)         elapsed: 64.79005 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
2019-01-08T08:42:13.344335 (delta: 0.03678)         elapsed: 64.826877 ******** 
changed: [127.0.0.1] => {"changed": true, "checksum": "4da6298723b4228f5cad23ca2155b6dd5f813feb", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "5822a4984246284d298ea2bfb437d54f", "mode": "0644", "owner": "root", "size": 423, "src": "/root/.ansible/tmp/ansible-tmp-1546936933.38-82950065144960/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
2019-01-08T08:42:14.365262 (delta: 1.020882)         elapsed: 65.847804 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.370099", "end": "2019-01-08 08:42:14.870131", "rc": 0, "start": "2019-01-08 08:42:14.500032", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: elasticsearch-deployment \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Pass ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: elasticsearch-deployment ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Pass "]}

TASK [Apply the litmus result CR] **********************************************
2019-01-08T08:42:14.908288 (delta: 0.542983)         elapsed: 66.39083 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.612148", "end": "2019-01-08 08:42:15.665106", "failed_when_result": false, "rc": 0, "start": "2019-01-08 08:42:15.052958", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/elasticsearch-deployment configured", "stdout_lines": ["litmusresult.litmus.io/elasticsearch-deployment configured"]}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=30   changed=20   unreachable=0    failed=0   

2019-01-08T08:42:15.694501 (delta: 0.786169)         elapsed: 67.177043 ******* 
=============================================================================== ```

gprasath commented on apps/elasticsearch/deployers/test.yml:
```diff
@@ -0,0 +1,73 @@
+                executable: /bin/bash
+
+         ## Deploying the application
+            - include_tasks: /common/utils/deploy_single_app.yml
```
Can you provide a comment about the deployer here.

gprasath commented on apps/elasticsearch/deployers/test_vars.yml:
```diff
@@ -0,0 +1,11 @@
+action: "{{ lookup('env','ACTION') }}"
+app_pvc: "{{ lookup('env','APP_PVC') }}"
+storage_class: "{{ lookup('env','PROVIDER_STORAGE_CLASS') }}"
+operator_ns: openebs
```
Where do we use operator_ns?

vibhor995 replied:
@gprasath we are using it while deletion / de-provisioning of application
